File: skills/binance-hunter-safe/scripts/price_collector.py
import urllib.request
import json
import sqlite3
import os

DB_DIR = os.path.join(os.path.dirname(__file__), '..', '..', '..', 'memory')
os.makedirs(DB_DIR, exist_ok=True)
DB_FILE = os.path.join(DB_DIR, 'price_warehouse.sqlite')

ASSETS = ["BTCUSDT", "ETHUSDT", "SOLUSDT", "TSLA"]

def get_binance_price(symbol):
    try:
        url = f"https://api.binance.com/api/v3/ticker/price?symbol={symbol}"
        req = urllib.request.Request(url, headers={'User-Agent': 'Mozilla/5.0'})
        with urllib.request.urlopen(req, timeout=5) as response:
            data = json.loads(response.read().decode())
            return float(data['price'])
    except Exception:
        return None

def main():
    conn = sqlite3.connect(DB_FILE)
    c = conn.cursor()
    c.execute('''CREATE TABLE IF NOT EXISTS asset_prices (
                    id INTEGER PRIMARY KEY AUTOINCREMENT,
                    asset TEXT NOT NULL,
                    price REAL NOT NULL,
                    timestamp_utc DATETIME DEFAULT CURRENT_TIMESTAMP
                 )''')
    c.execute('CREATE INDEX IF NOT EXISTS idx_asset ON asset_prices(asset)')
    
    count = 0
    for asset in ASSETS:
        if asset.endswith("USDT"):
            px = get_binance_price(asset)
            if px is not None:
                c.execute("INSERT INTO asset_prices (asset, price) VALUES (?, ?)", (asset, px))
                count += 1
                
    conn.commit()
    conn.close()
    if count > 0:
        print(f"Guardados {count} precios correctamente en el Warehouse.")
    else:
        print("No se pudo extraer ningun precio.")

if __name__ == "__main__":
    main()

File: skills/binance-hunter-safe/scripts/price_collector.mjs
#!/usr/bin/env node
import fs from 'fs';
import path from 'path';

const DB_DIR = path.join(path.dirname(new URL(import.meta.url).pathname).replace(/^\/([A-Za-z]:)/, '$1'), '..', '..', '..', 'memory');
if (!fs.existsSync(DB_DIR)) {
  fs.mkdirSync(DB_DIR, { recursive: true });
}
const DB_FILE = path.join(DB_DIR, 'price_warehouse.csv');

const ASSETS = ["BTCUSDT", "ETHUSDT", "SOLUSDT", "TSLA"]; 

async function getBinancePrice(symbol) {
  try {
     const res = await fetch(`https://api.binance.com/api/v3/ticker/price?symbol=${symbol}`);
     if(!res.ok) return null;
     const data = await res.json();
     return parseFloat(data.price);
  } catch(e) {
     return null;
  }
}

async function collect() {
  let isNew = !fs.existsSync(DB_FILE);
  if (isNew) {
      fs.writeFileSync(DB_FILE, "timestamp_utc,asset,price\n", "utf8");
  }
  
  let count = 0;
  const now = new Date().toISOString();
  
  for (const asset of ASSETS) {
     if(asset.endsWith("USDT")) {
         const px = await getBinancePrice(asset);
         if(px !== null) {
            fs.appendFileSync(DB_FILE, `${now},${asset},${px}\n`, "utf8");
            count++;
         }
     }
  }
  
  console.log(`Guardados ${count} precios correctamente en el Warehouse (CSV).`);
}

collect();
